docs/developer-docs/smart-contracts/deploy/custom-testnets.mdx

```diff
@@ -0,0 +1,217 @@
+
+To provide a testnet-like environment, ICP has a smart contract [playground](https://m7sm4-2iaaa-aaaab-qabra-cai.ic0.app/) network that can be used for small-scale, temporary testing. Canisters deployed to the playground are restricted to certain parameters. These limitations are:
+
+- Cycle transfer instructions are silently ignored by the playground.
+- Canisters can use at most 1GiB of memory.
+- Canisters can call the management canister to manage itself without being the controller.
+- Deployed canisters expire after 20 minutes. Upgrades to canisters reset this timer. When the timer runs out, the canister(s) will be uninstalled.
+- Wasm files can't be gzipped.
+- Wasm files will be analyzed to remove any potentially expensive or malicious operations.
+
+For advanced developers and use cases that want to use a testnet environment without these restrictions, there are two options:
+
+- **Private testnets**: Deploy your own instance of the playground, allowing you to control the playground settings, remove restrictions, and modify the canister pool. This option costs cycles, as you will need to fund your instance of the playground with cycles that will be used to fund canisters deployed to your playground.
+
+- **Synthetic testnets**: Use the `dfx` named network feature to create a local network that is segmented away from the default local network created by `dfx`.
+
+## Private testnet
+
+To create a private testnet, you can create a custom instance of the playground. Using your own instance of the playground allows for extensive customization, such as:
+
+- Enabling access control by restricting the playground's usage to only allow certain principals.
+
+- Configuring more generous canister timeouts and the number of available cycles.
+
+- Allowing some or all of the functions that the public playground does not allow, such as sending cycles to other canisters.
+
+Using a custom playground can also help simplify development for teams, since the whole team can use a custom playground without needing to manage individual cycle balances.
+
+### Creating a custom playground instance
+
+- #### Step 1: Clone the playground repo with the command:
+
+```
+git clone https://github.com/dfinity/motoko-playground
+```
+
+- #### Step 2: To create a separate pool, first use the current playground [pool and `wasm-utils` canisters as the starting point](https://github.com/dfinity/motoko-playground/tree/main/service).
+
+- #### Step 3: Then, edit the `pool/Main.mo` file to change your custom playground settings.
+
+Some settings you may want to change are:
+
+- Add [access control](https://github.com/dfinity/motoko-playground/blob/main/service/pool/Main.mo#L134) as desired, such as creating a `whitelist` of principals that are permitted to use the custom playground.
```

Comment:
`whiteList` is used here and `allowList` is used in the paragraph below

Reply:
Fixed so both use `allowList`